pytest_operator/plugin.py

```diff
@@ -116,6 +124,16 @@ def pytest_collection_modifyitems(session, config, items):
             item.add_marker("asyncio")
 
 
+def _source_charm_dir(path: Path) -> typing.Optional[Path]:
+    """Return path to charm source directory. (../TOX_ENV_DIR)."""
+    if ".tox" not in path.parts:
+        log.warning("could not find `.tox` directory in path %s", path)
+        return None
+
+    tox_env_dir_index = path.parts.index(".tox")
+    return Path(*path.parts[:tox_env_dir_index])
```

Comment:
Is *this* the intent?

```suggestion
    """Return path to charm source directory. ($TOX_ENV_DIR/../..)."""
    return Path(os.environ["TOX_ENV_DIR"]).parent.parent
```

Reply:
I was wondering about this in my review, as well, and commented on the call to it below.  I'm leaning toward including an explicit `--crash-dump-output` option which defaults to the current directory, since I think that is probably safer and easier than any heuristic.